### DISCUSSION TITLE
Compiling grblhal on platformio using raspberry PI Pico

### DISCUSSION THREAD

**Discussion by CHTUZKI** [grblHAL/RP2040#30] [General]
Hi.
I want to use PlatformIO IDE to compile grblHAL, but I have a lot of problems, I don't know if this idea is feasible, can someone give me some advice.
I see that PlatformIO only provides framework for arduino. The default arduino framework of PlatformIO seems to be written directly according to the registers of the RP2040. I found another arduino framework on github written according to the pico SDK. I've tried both frameworks and it wasn't that easy.I don't know if I'm right，😁😁😁

**Comment by terjeio** [comment]
The driver uses the Pico SDK so at least that has to be available for it to compile.
So you have to persuade PlatformIO to use the SDK version of the Arduino framework or write a new driver.
